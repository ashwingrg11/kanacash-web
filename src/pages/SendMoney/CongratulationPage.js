import React from "react";
import { Row, Col, Container } from "react-bootstrap";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import PropTypes from "prop-types";
import Confetti from "react-confetti";
import ROUTES from "../../assets/Routes/Routes";
import CongratulationImg from "../../assets/images/congratulation.svg";
import Header from "../../containers/layout/Header";
import Footer from "../../containers/layout/Footer";
import { clearTransactionData } from "../../store/actions/TrasactionAction";
import Disclaimer from "../../components/Disclaimer/Disclaimer";
class CongratulationPage extends React.Component {
  render() {
    const receiverInfo = this.props.location.state;
    return (
      <React.Fragment>
        <Header />
        <div className="shadow-line"></div>
        <section className="front-background-no-bg">
          <Container>
            <Row>
              <Col className="col-md-6 offset-md-3 d-flex justify-content-center mt-5 mb-4">
                <Row>
                  <Col className="d-flex  flex-column align-items-center justify-align-center">
                    <Confetti
                      recycle={false}
                      width={"300px"}
                      height={"400px"}
                      numberOfPieces={200}
                      style={{ margin: "0 auto" }}
                      drawShape={(ctx) => {
                        ctx.beginPath();
                        for (let i = 0; i < 22; i++) {
                          const angle = 0.2 * i;
                          const x = (0.2 + 1.5 * angle) * Math.cos(angle);
                          const y = (0.2 + 1.5 * angle) * Math.sin(angle);
                          ctx.lineTo(x, y);
                        }
                        ctx.stroke();
                        ctx.closePath();
                      }}
                      opacity={3.0}
                    />{" "}
                    <img
                      src={CongratulationImg}
                      alt="CongratulationImg"
                      className="mt-2"
                    />
                    <div className="mb-1 justify-content-center text-center">
                      <h2 className="sub-title title-success mb-1">
                        {" "}
                        Congratulations!
                      </h2>
                      <p>Your money transfer is on its way to</p>
                      <span className="account-title mb-3 d-block">
                        {receiverInfo?.first_name} {receiverInfo?.last_name}
                      </span>
                      <p>
                        {" "}
                        Please go to your dashboard to view your receipt and
                        other details. We will send you an email with your
                        recipt and notification agter the transaction is
                        delivered
                      </p>
                    </div>
                    <div className="button-group text-center">
                      <button
                        className="btn btn-secondary m-2"
                        onClick={() => {
                          this.props.history.push(`${ROUTES.DASHBOARD}`);
                        }}
                      >
                        Go to Dashboard
                      </button>
                      <button className="btn btn-primary btn-primary-border m-2">
                        Notify Beneficary
                      </button>
                    </div>
                  </Col>
                </Row>
              </Col>
            </Row>
          </Container>
        </section>
        <Footer />
        {!localStorage.acceptPolicy && <Disclaimer />}
      </React.Fragment>
    );
  }
}

CongratulationPage.propTypes = {
  receiverInfo: PropTypes.object,
  history: PropTypes.object,
  location: PropTypes.object,
};

const mapStateToProps = (state) => ({
  receiverInfo: state.transaction.receiverInfo,
  acceptPolicy: state.auth.accept_policy,
});

const mapActionToProps = { clearTransactionData };

export default connect(
  mapStateToProps,
  mapActionToProps
)(withRouter(CongratulationPage));
