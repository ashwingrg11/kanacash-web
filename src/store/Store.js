import { createStore, combineReducers, applyMiddleware, compose } from "redux";
// import { composeWithDevTools } from 'redux-devtools-extension';
import Thunk from "redux-thunk";
import LoginReducer from "./reducers/Login";
import ForgotPasswordReducer from "./reducers/ForgotPassword";
import LoaderReducer from "./reducers/Loader";
import ModalReducer from "./reducers/Modal";
import NetworkReducer from "./reducers/Network";
import AuthReducer from "./reducers/Auth";
import ErrorReducer from "./reducers/Error";
import MiscellaneousReducer from "./reducers/MiscellaneousReducer";
import senderDetailReducers from "./reducers/SenderDetailReducer";
import TransactionReducer from "./reducers/TransactionReducer";
import SignUpReducer from "./reducers/SignUpReducer";
import AdminReducers from "./reducers/adminReducers";

const middleware = [Thunk];
const initialState = {};

const composeEnhancers =
  typeof window === "object" && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        trace: true,
      })
    : compose;

const rootReducer = combineReducers({
  login: LoginReducer,
  forgotPassword: ForgotPasswordReducer,
  loader: LoaderReducer,
  modal: ModalReducer,
  network: NetworkReducer,
  auth: AuthReducer,
  error: ErrorReducer,
  miscellaneous: MiscellaneousReducer,
  senderDetail: senderDetailReducers,
  transaction: TransactionReducer,
  signup: SignUpReducer,
  admin: AdminReducers,
});

const store = createStore(
  rootReducer,
  initialState,
  composeEnhancers(applyMiddleware(...middleware))
);
export default store;
