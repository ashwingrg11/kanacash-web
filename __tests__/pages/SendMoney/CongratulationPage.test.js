import React from "react";
import CongratulationPage from "../../../src/pages/SendMoney/CongratulationPage";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import { shallow } from "enzyme";
import store from "../../../src/store/Store";

import renderer from "react-test-renderer";
describe("<CongratulationPage />", () => {
  it("CongratulationPage renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <CongratulationPage />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("CongratulationPage Row Component", () => {
    const foo = "hello";
    const bar = "hello";
    expect(foo).toEqual(bar);
  });
});
