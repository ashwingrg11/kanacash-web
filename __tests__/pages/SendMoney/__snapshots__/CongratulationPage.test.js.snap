// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CongratulationPage /> CongratulationPage renders correctly 1`] = `
Array [
  <header
    className="header-main"
  >
    <div
      className="container"
    >
      <nav
        className="navbar navbar-expand-lg navbar-light bg-light"
      >
        <div
          className="navbar-brand"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <img
            alt="logo"
            className="logo"
            src=""
          />
        </div>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          className="navbar-toggler collapsed"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          aria-expanded={null}
          className="navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            className="ml-md-auto navbar-nav"
            onKeyDown={[Function]}
          >
            <span
              className="btn btn-primary btn-theme btn-primary-border"
              onClick={[Function]}
            >
              Sign in
            </span>
            <button
              className="btn btn-primary btn-theme btn-primary-theme"
              onClick={[Function]}
            >
              Sign up
            </button>
          </div>
        </div>
      </nav>
    </div>
  </header>,
  <div
    className="shadow-line"
  />,
  <section
    className="front-background-no-bg"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-6 offset-md-3 d-flex justify-content-center mt-5 mb-4 col"
        >
          <div
            className="row"
          >
            <div
              className="d-flex  flex-column align-items-center justify-align-center col"
            >
              <canvas
                height="400px"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "margin": "0 auto",
                    "pointerEvents": "none",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "zIndex": 2,
                  }
                }
                width="300px"
              />
               
              <img
                alt="CongratulationImg"
                className="mt-2"
                src=""
              />
              <div
                className="mb-1 justify-content-center text-center"
              >
                <h2
                  className="sub-title title-success mb-1"
                >
                   
                  Congratulations!
                </h2>
                <p>
                  Your money transfer is on its way to
                </p>
                <span
                  className="account-title mb-3 d-block"
                >
                   
                </span>
                <p>
                   
                  Please go to your dashboard to view your receipt and other details. We will send you an email with your recipt and notification agter the transaction is delivered
                </p>
              </div>
              <div
                className="button-group text-center"
              >
                <button
                  className="btn btn-secondary m-2"
                  onClick={[Function]}
                >
                  Go to Dashboard
                </button>
                <button
                  className="btn btn-primary btn-primary-border m-2"
                >
                  Notify Beneficary
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <footer
    className="footer-section text-center"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div
            className="foot-nav"
          >
            <a
              aria-current="page"
              className="link active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Home
            </a>
            <a
              aria-current={null}
              className="link"
              href="/about-us"
              onClick={[Function]}
            >
              About Us
            </a>
            <a
              aria-current={null}
              className="link"
              href="/terms-of-service"
              onClick={[Function]}
            >
              Terms
            </a>
            <a
              aria-current={null}
              className="link"
              href="/privacy-policy"
              onClick={[Function]}
            >
              Privacy Policy
            </a>
            <a
              aria-current={null}
              className="link"
              href="/copyright"
              onClick={[Function]}
            >
              Copyright
            </a>
            <a
              aria-current={null}
              className="link"
              href="/contact-us"
              onClick={[Function]}
            >
              Contact
            </a>
          </div>
        </div>
        <div
          className="copyright col-sm-12"
        >
          <p>
            © Golden Money Transfer Inc. 2020. All Rights Reserved.
             
            <b>
              Licensed by
            </b>
             Department of Financial Institutions, Nevada
          </p>
        </div>
      </div>
    </div>
  </footer>,
  <div
    className="disclaimer text-center"
  >
    <div
      className="container"
    >
      <div
        className="policy"
      >
         
        This website uses cookies to ensure you get the best experience on our website, as described in our
         
        <span
          className="btn-link"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
           
            privacy policy.
        </span>
      </div>
      <div
        className="policy1"
      >
        This website uses cookies,
        <span
          className="btn-link"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
           
            learn more.
        </span>
      </div>
      <div
        className="buttonDiv"
      >
        <button
          className="btn-outline-warning"
          onClick={[Function]}
          style={
            Object {
              "border": "none",
              "borderRadius": "50px",
              "color": "black",
              "padding": "5px 15px",
            }
          }
        >
          Accept
        </button>
      </div>
    </div>
  </div>,
]
`;
